docs/concepts/callbacks.md

```diff
@@ -1,2 +1,226 @@
+           INFO     Building the artifact ...                                                            __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                                       __main__.py:231
+```
+
+The user-specified parameters are:
+
+- `min_delta`: The minimum amount of improvement that a model must have over the previous best model in order for fine-tuning to continue. By default, this is zero, meaning that the training will not stop early unless the performance starts to decrease.
+- `patience`: The number of consecutive rounds without improvement before the training is stopped, set to two by default.
+- `baseline`: an optional parameter against which the model's score is compared when checking for improvement, instead of the best previous model. If specified, the improvement every epoch is measured by comparison to this value instead of the best-performing model so far.
+
+In code:
+
+```python
+import finetuner
+from finetuner.callback import EarlyStopping
+
+run = finetuner.fit(
+    ...,
+    callbacks= [
+        EarlyStopping(
+            monitor = "train_loss",
+            mode = "min",
+            patience=2,
+            min_delta=1,
+            baseline=1.5
+        )
+    ]
+)
+```
+
+## WiSEFTCallback
+
+WiSE-FT, proposed by Mitchell et al. in [Robust fine-tuning of zero-shot models](https://arxiv.org/abs/2109.01903),
+has proven to be an effective way of fine-tuning models with strong zero-shot capability,
+such as CLIP:
+
+> Large pre-trained models such as CLIP or ALIGN offer consistent accuracy across a range of data distributions when performing zero-shot inference (i.e., without fine-tuning on a specific dataset). Although existing fine-tuning methods substantially improve accuracy on a given target distribution, they often reduce robustness to distribution shifts. We address this tension by introducing a simple and effective method for improving robustness while fine-tuning: ensembling the weights of the zero-shot and fine-tuned models (WiSE-FT).
+
+```python
+import finetuner
+from finetuner.callback import WiSEFTCallback
+
+run = finetuner.fit(
+    model='clip-base-en',
+    ...,
+    loss='CLIPLoss',
+    callbacks=[WiSEFTCallback(alpha=0.5)]
+)
+```
+
+Alpha should be assigned a value that is greater than or equal to 0 but less than or equal to 1:
+
+1. When alpha is set to 0, the fine-tuned model is the same as the pre-trained model.
+2. When alpha is set to 1, the pre-trained weights are not used.
+3. When alpha is a floating-point number between 0 and 1, we combine the weights of the pre-trained and fine-tuned models.
+
+```{warning}
+It is recommended to use WiSEFTCallback when fine-tuning CLIP.
+We can not ensure it works for other types of models, such as ResNet or BERT.
+
+Please refer to {ref}`Apply WiSE-FT <wise-ft>` in the CLIP fine-tuning example for more information.
```

Comment:
This ref seems not to work. It looks like the URL it should resolve to is: https://finetuner.jina.ai/notebooks/text_to_image/#advanced-wise-ft